### DISCUSSION TITLE
How to load linear EXR images?

### DISCUSSION THREAD

**Discussion by mcg222** [colour-science/colour-checker-detection#77] [Q&A]
To import images for colour-checker-detection, I use:

COLOUR_CHECKER_IMAGE_PATHS = glob.glob(
    os.path.join('/content/', '*.png'))

COLOUR_CHECKER_IMAGES = [
    colour.cctf_decoding(colour.io.read_image(path),function="REDLog")
    for path in COLOUR_CHECKER_IMAGE_PATHS
]

for image in COLOUR_CHECKER_IMAGES:
    colour.plotting.plot_image(colour.cctf_encoding(image));

I'd like to be able to import linear EXR images.  How can I do that?    I enabled the OpenCV option to allow for EXR images, but when I bring them in they show up as black.  Is there some kind of pass through option so that images aren't linearized? 

Thanks!

**Comment by KelSolaar** [comment]
Hi @mcg222,

You will need to load the `freeimage` backend to have EXR read and write support. This is how we typically do that:

`python -c "import imageio;imageio.plugins.freeimage.download()"`

**Reply by mcg222** [reply]
Thanks!  Now the EXR images open!  The next thing I need to do is to bypass the encoding to linear and decoding to sRGB in:

COLOUR_CHECKER_IMAGES = [
colour.cctf_decoding(colour.io.read_image(path),function="Log3G10")
for path in COLOUR_CHECKER_IMAGE_PATHS
]

for image in COLOUR_CHECKER_IMAGES:
colour.plotting.plot_image(colour.cctf_encoding(image));

As it is taking my already linear EXR image, and attempting to linearize it.  With the above it thinks its Log3G10.  How can I tell it it's already linear?  Thanks!

**Reply by KelSolaar** [reply]
If your images are already in a linear state this should be enough:

```python
COLOUR_CHECKER_IMAGES = [
    colour.io.read_image(path) for path in COLOUR_CHECKER_IMAGE_PATHS
]
```

**Reply by mcg222** [reply]
@KelSolaar Thanks so much.  That worked perfectly!
